code/game/objects/items/grenades/atmos_grenades.dm

```diff
@@ -0,0 +1,82 @@
+/obj/item/grenade/gas_crystal
+	desc = "Some kind of crystal, this shouldn't spawn"
+	name = "Gas Crystal"
+	icon = 'icons/obj/grenade.dmi'
+	icon_state = "bluefrag"
+	inhand_icon_state = "flashbang"
+	resistance_flags = FIRE_PROOF
+
+
+/obj/item/grenade/gas_crystal/x_crystal
+	name = "X Crystal"
+	desc = "A crystal made from the Gas X, it's cold to the touch."
+	icon_state = "crystal_x"
+	var/stamina_damage = 30
+	var/freeze_range = 4
+
+/obj/item/grenade/gas_crystal/y_crystal
+	name = "Y Crystal"
+	desc = "A crystal made from the Gas Y, you can see the liquid gases inside."
+	icon_state = "crystal_y"
+	var/refill_range = 5
+
+/obj/item/grenade/gas_crystal/z_crystal
+	name = "Z Crystal"
+	desc = "A crystal made from the Gas Z, you can see the liquid plasma inside."
+	icon_state = "crystal_z"
+	ex_dev = 1
+	ex_heavy = 2
+	ex_light = 4
+	ex_flame = 2
+
+/obj/item/grenade/gas_crystal/preprime(mob/user, delayoverride, msg = TRUE, volume = 60)
+	var/turf/T = get_turf(src)
+	log_grenade(user, T) //Inbuilt admin procs already handle null users
+	if(user)
+		add_fingerprint(user)
+		if(msg)
+			to_chat(user, "<span class='warning'>You crush the [src]! [capitalize(DisplayTimeText(det_time))]!</span>")
+	if(shrapnel_type && shrapnel_radius)
+		shrapnel_initialized = TRUE
+		AddComponent(/datum/component/pellet_cloud, projectile_type=shrapnel_type, magnitude=shrapnel_radius)
+	active = TRUE
+	icon_state = initial(icon_state) + "_active"
+	playsound(src, 'sound/effects/hit_on_shattered_glass.ogg', volume, TRUE)
+	SEND_SIGNAL(src, COMSIG_GRENADE_ARMED, det_time, delayoverride)
+	addtimer(CALLBACK(src, .proc/prime), isnull(delayoverride)? det_time : delayoverride)
+
+/obj/item/grenade/gas_crystal/x_crystal/prime(mob/living/lanced_by)
+	. = ..()
+	update_mob()
+	playsound(src, 'sound/effects/spray2.ogg', 100, TRUE)
+	for(var/turf/T in view(freeze_range,loc))
+		if(isopenturf(T))
+			var/turf/open/F = T
+			if(F.air.temperature > 260)
+				F.air.temperature = 200
```

Comment:
Might be good to not hardcode the temp change, instead adding some pretty cold inert gas. I shouldn't be able to neutralize a fusion flood in a space with one gem

Reply:
true, what would be the way to do this? because i know how to add something like n2, i know how to change the temp of air, but i dont know how to add cold n2

Reply:
`T.atmos_spawn_air("some gas string")` will do what you want, it makes a new gasmix with the required contents and then merges the turfs mix and it's own, allowing for temperature share. Up to you how you use it, format is `"gasid=count;infinitegasids=count;TEMP=temperature"`